docs/examples/day-night-masking.py

```diff
@@ -0,0 +1,197 @@
+"""
+Day-Night Masking
+=================
+
+Masking day-night periods using the PVAnalytics daytime module.
+"""
+
+# %%
+# Identifying and masking day-night periods in an AC power time series or
+# irradiance time series can aid in future data analysis, such as detecting
+# if a time series has daylight savings time or time shifts.
+
+import pvanalytics
+from pvanalytics.features.daytime import power_or_irradiance
+import matplotlib.pyplot as plt
+import pandas as pd
+import pathlib
+import pvlib
+import numpy as np
+
+# %%
+# First, read in the 1-minute sampled AC power time series data, taken
+# from the SERF East installation on the NREL campus.
+# This sample is provided from the NREL PVDAQ database, and contains
+# a column representing an AC power data stream.
+
+pvanalytics_dir = pathlib.Path(pvanalytics.__file__).parent
+ac_power_file = pvanalytics_dir / 'data' / 'serf_east_1min_ac_power.csv'
+data = pd.read_csv(ac_power_file, index_col=0, parse_dates=True)
+data = data.sort_index()
+# This is the known frequency of the time series. You may need to infer
+# the frequency or set the frequency with your AC power time series.
+freq = "1T"
+# These are the latitude-longitude coordinates associated with the
+# SERF East system.
+latitude = 39.742
+longitude = -105.173
+# Plot the time series.
+data['ac_power__752'].plot()
+plt.xticks(rotation=20)
+plt.xlabel("Date")
+plt.ylabel("AC Power (kW)")
+plt.tight_layout()
+plt.show()
+
+# %%
+# The pvlib.solarposition.sun_rise_set_transit_spa function is
+# used to get sunrise and sunset times for each day at the site location, and
+# the daytime mask is calculated based on these times. Data associated with
+# day time periods is labeled as True, and data associated with nighttime
+# periods is labeled as False.
```

Comment:
Probably worth pointing out that this is only done for comparison; "true" sunrise and sunset times are not needed to run `pvanalytics.features.daytime.power_or_irradiance`.  Could even move the SPA calculation and put it after the plot.

Reply:
@kanderso-nrel I added some additional comments on this being for comparison. I did move the script around a little, but this is still before the plot as we use SPA sunrise and sunset in the plot (they are represented by the vertical lines)